### ISSUE TITLE
Crun fails to clean old content of descriptors.json

### ISSUE THREAD

**Issue by marusak** [cockpit-project/bots#2654]
Upstream issue https://github.com/containers/crun/issues/787

**Comment by cockpituous** [comment]
fedora-35
Ooops, it happened again<hr>
```
# ----------------------------------------------------------------------
# testCheckpointRestoreCGroupsV2 (__main__.TestApplication)
Warning: Stopping podman.service, but it can still be activated by:
  podman.socket

DevTools listening on ws://127.0.0.1:9904/devtools/browser/9d139585-426e-4db6-9f52-84c9a28cb37a
[1123/104008.293249:ERROR:egl_util.cc(74)] Failed to load GLES library: /usr/lib64/chromium-browser/swiftshader/libGLESv2.so: /usr/lib64/chromium-browser/swiftshader/libGLESv2.so: cannot open shared object file: No such file or directory
[1123/104008.295079:ERROR:viz_main_impl.cc(162)] Exiting GPU process due to errors during initialization
[1123/104008.297754:ERROR:gpu_init.cc(441)] Passthrough is not supported, GL is disabled
> error: Modal: Specify at least one of: title, aria-label, aria-labelledby.
CDP: {"source":"network","level":"error","text":"Failed to load resource: the server responded with a status of 404 (Not Found)","timestamp":1637664009555.6182,"url":"http://127.0.0.2:9191/cockpit/$c2b31d3f544eb2d56972a24a095dfe13241d8b5407fc5bdfb1eeffde0b03ba7e/podman/po.js","networkRequestId":"8658.67"}
CDP: {"source":"network","level":"error","text":"Failed to load resource: the server responded with a status of 404 (Not Found)","timestamp":1637664009780.831,"url":"http://127.0.0.2:9191/cockpit/$c2b31d3f544eb2d56972a24a095dfe13241d8b5407fc5bdfb1eeffde0b03ba7e/podman/po.js","networkRequestId":"8658.90"}
> warning: grep: /sys/class/dmi/id/power/autosuspend_delay_ms: Input/output error

time="2021-11-23T10:40:10Z" level=warning msg="The input device is not a TTY. The --tty and --interactive flags might not work properly"
Traceback (most recent call last):
  File "/usr/lib64/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/work/bots/make-checkout-workdir/test/check-application", line 959, in testCheckpointRestoreCGroupsV2
    return self._testCheckpointRestore()
  File "/work/bots/make-checkout-workdir/test/check-application", line 1054, in _testCheckpointRestore
    b.wait(lambda: self.getContainerAttr("swamped-crate", "State") in 'running')
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 401, in wait
    raise Error('timed out waiting for predicate to become true')
testlib.Error: timed out waiting for predicate to become true
Wrote screenshot to TestApplication-testCheckpointRestoreCGroupsV2-fedora-35-127.0.0.2-2301-FAIL.png
Wrote HTML dump to TestApplication-testCheckpointRestoreCGroupsV2-fedora-35-127.0.0.2-2301-FAIL.html
Wrote JS log to TestApplication-testCheckpointRestoreCGroupsV2-fedora-35-127.0.0.2-2301-FAIL.js.log
Journal extracted to TestApplication-testCheckpointRestoreCGroupsV2-fedora-35-127.0.0.2-2301-FAIL.log.gz
Warning: Stopping podman.service, but it can still be activated by:
  podman.socket
> warning: transport closed: disconnected
Failed to get session path: No session '66' known
Traceback (most recent call last):
  File "/work/bots/make-checkout-workdir/test/check-application", line 959, in testCheckpointRestoreCGroupsV2
    return self._testCheckpointRestore()
  File "/work/bots/make-checkout-workdir/test/check-application", line 1054, in _testCheckpointRestore
    b.wait(lambda: self.getContainerAttr("swamped-crate", "State") in 'running')
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 401, in wait
    raise Error('timed out waiting for predicate to become true')
testlib.Error: timed out waiting for predicate to become true

# Result testCheckpointRestoreCGroupsV2 (__main__.TestApplication) failed
# 1 TEST FAILED [92s on 4-cockpit-10]
not ok 4 /work/bots/make-checkout-workdir/test/check-application TestApplication.testCheckpointRestoreCGroupsV2 [ND@1]
```
First occurrence: 2021-11-23T10:41:38.324713 | revision 4aad10d6a28f74da48617a00b28c4a79a8d09861
Times recorded: 1
Latest occurrences:

- 2021-11-23T10:41:38.324713 | revision 4aad10d6a28f74da48617a00b28c4a79a8d09861

**Comment by cockpituous** [comment]
rhel-9-0
Ooops, it happened again<hr>
```
# ----------------------------------------------------------------------
# testCheckpointRestoreCGroupsV2 (__main__.TestApplication)
Warning: Stopping podman.service, but it can still be activated by:
  podman.socket

DevTools listening on ws://127.0.0.1:9374/devtools/browser/0ab7ca69-b6fe-4eda-8b46-28d4fb454721
[1123/111321.453359:ERROR:egl_util.cc(74)] Failed to load GLES library: /usr/lib64/chromium-browser/swiftshader/libGLESv2.so: /usr/lib64/chromium-browser/swiftshader/libGLESv2.so: cannot open shared object file: No such file or directory
[1123/111321.456319:ERROR:viz_main_impl.cc(162)] Exiting GPU process due to errors during initialization
[1123/111321.460070:ERROR:gpu_init.cc(441)] Passthrough is not supported, GL is disabled
CDP: {"source":"security","level":"error","text":"Refused to apply inline style because it violates the following Content Security Policy directive: \"default-src 'self' http://127.0.0.2:9091\". Either the 'unsafe-inline' keyword, a hash ('sha256-BMIkowAqVEI88ibUJ1NtYB8ByZOuS4MTi73eGoFo2+Q='), or a nonce ('nonce-...') is required to enable inline execution. Note that hashes do not apply to event handlers, style attributes and javascript: navigations unless the 'unsafe-hashes' keyword is present. Note also that 'style-src' was not explicitly set, so 'default-src' is used as a fallback.\n","timestamp":1637666002884.719,"url":"http://127.0.0.2:9091/cockpit/$46f46d72a0d3df35f7294a3c40ce32baef4d4dd97dbd7bb7cd9447e3f3c915de/shell/index.js","lineNumber":1,"stackTrace":{"callFrames":[{"functionName":"ke","scriptId":"24","url":"http://127.0.0.2:9091/cockpit/$46f46d72a0d3df35f7294a3c40ce32baef4d4dd97dbd7bb7cd9447e3f3c915de/shell/index.js","lineNumber":1,"columnNumber":121721},{"functionName":"Ze","scriptId":"24","url":"http://127.0.0.2:9091/cockpit/$46f46d72a0d3df35f7294a3c40ce32baef4d4dd97dbd7bb7cd9447e3f3c915de/shell/index.js","lineNumber":1,"columnNumber":130758},{"functionName":"append","scriptId":"24","url":"http://127.0.0.2:9091/cockpit/$46f46d72a0d3df35f7294a3c40ce32baef4d4dd97dbd7bb7cd9447e3f3c915de/shell/index.js","lineNumber":1,"columnNumber":132439},{"functionName":"D","scriptId":"24","url":"http://127.0.0.2:9091/cockpit/$46f46d72a0d3df35f7294a3c40ce32baef4d4dd97dbd7bb7cd9447e3f3c915de/shell/index.js","lineNumber":1,"columnNumber":349379},{"functionName":"","scriptId":"24","url":"http://127.0.0.2:9091/cockpit/$46f46d72a0d3df35f7294a3c40ce32baef4d4dd97dbd7bb7cd9447e3f3c915de/shell/index.js","lineNumber":1,"columnNumber":358581},{"functionName":"","scriptId":"24","url":"http://127.0.0.2:9091/cockpit/$46f46d72a0d3df35f7294a3c40ce32baef4d4dd97dbd7bb7cd9447e3f3c915de/shell/index.js","lineNumber":1,"columnNumber":643600},{"functionName":"","scriptId":"24","url":"http://127.0.0.2:9091/cockpit/$46f46d72a0d3df35f7294a3c40ce32baef4d4dd97dbd7bb7cd9447e3f3c915de/shell/index.js","lineNumber":1,"columnNumber":643604}]}}
> error: Modal: Specify at least one of: title, aria-label, aria-labelledby.
CDP: {"source":"network","level":"error","text":"Failed to load resource: the server responded with a status of 404 (Not Found)","timestamp":1637666003173.585,"url":"http://127.0.0.2:9091/cockpit/$46f46d72a0d3df35f7294a3c40ce32baef4d4dd97dbd7bb7cd9447e3f3c915de/podman/po.js","networkRequestId":"5215.67"}
CDP: {"source":"network","level":"error","text":"Failed to load resource: the server responded with a status of 404 (Not Found)","timestamp":1637666003310.852,"url":"http://127.0.0.2:9091/cockpit/$46f46d72a0d3df35f7294a3c40ce32baef4d4dd97dbd7bb7cd9447e3f3c915de/podman/po.js","networkRequestId":"5215.89"}
> warning: grep: /sys/class/dmi/id/power/autosuspend_delay_ms: Input/output error

time="2021-11-23T06:13:23-05:00" level=warning msg="The input device is not a TTY. The --tty and --interactive flags might not work properly"
Traceback (most recent call last):
  File "/usr/lib64/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/work/bots/make-checkout-workdir/test/check-application", line 959, in testCheckpointRestoreCGroupsV2
    return self._testCheckpointRestore()
  File "/work/bots/make-checkout-workdir/test/check-application", line 1054, in _testCheckpointRestore
    b.wait(lambda: self.getContainerAttr("swamped-crate", "State") in 'running')
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 401, in wait
    raise Error('timed out waiting for predicate to become true')
testlib.Error: timed out waiting for predicate to become true
Wrote screenshot to TestApplication-testCheckpointRestoreCGroupsV2-rhel-9-0-127.0.0.2-2201-FAIL.png
Wrote HTML dump to TestApplication-testCheckpointRestoreCGroupsV2-rhel-9-0-127.0.0.2-2201-FAIL.html
Wrote JS log to TestApplication-testCheckpointRestoreCGroupsV2-rhel-9-0-127.0.0.2-2201-FAIL.js.log
Journal extracted to TestApplication-testCheckpointRestoreCGroupsV2-rhel-9-0-127.0.0.2-2201-FAIL.log.gz
Warning: Stopping podman.service, but it can still be activated by:
  podman.socket
> warning: transport closed: disconnected
Failed to get session path: No session '30' known
Traceback (most recent call last):
  File "/work/bots/make-checkout-workdir/test/check-application", line 959, in testCheckpointRestoreCGroupsV2
    return self._testCheckpointRestore()
  File "/work/bots/make-checkout-workdir/test/check-application", line 1054, in _testCheckpointRestore
    b.wait(lambda: self.getContainerAttr("swamped-crate", "State") in 'running')
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 401, in wait
    raise Error('timed out waiting for predicate to become true')
testlib.Error: timed out waiting for predicate to become true

# Result testCheckpointRestoreCGroupsV2 (__main__.TestApplication) failed
# 1 TEST FAILED [92s on 3-ci-srv-01]
not ok 4 /work/bots/make-checkout-workdir/test/check-application TestApplication.testCheckpointRestoreCGroupsV2 [ND@0]
```
First occurrence: 2021-11-23T11:14:51.944511 | revision cb9438497c6042913af1da90966f01b5f120fc7b
Times recorded: 1
Latest occurrences:

- 2021-11-23T11:14:51.944511 | revision cb9438497c6042913af1da90966f01b5f120fc7b

**Comment by cockpituous** [comment]
fedora-34
Ooops, it happened again<hr>
```
# ----------------------------------------------------------------------
# testCheckpointRestoreCGroupsV2 (__main__.TestApplication)
Warning: Stopping podman.service, but it can still be activated by:
  podman.socket

DevTools listening on ws://127.0.0.1:9542/devtools/browser/6c277b58-b9d0-4a59-9544-996bc7a6b8ad
[1123/110213.181930:ERROR:egl_util.cc(74)] Failed to load GLES library: /usr/lib64/chromium-browser/swiftshader/libGLESv2.so: /usr/lib64/chromium-browser/swiftshader/libGLESv2.so: cannot open shared object file: No such file or directory
[1123/110213.184361:ERROR:viz_main_impl.cc(162)] Exiting GPU process due to errors during initialization
[1123/110213.187487:ERROR:gpu_init.cc(441)] Passthrough is not supported, GL is disabled
> error: Modal: Specify at least one of: title, aria-label, aria-labelledby.
CDP: {"source":"network","level":"error","text":"Failed to load resource: the server responded with a status of 404 (Not Found)","timestamp":1637665334898.292,"url":"http://127.0.0.2:9191/cockpit/$5d7b4a773812d7654b86f8c74ff42a11c495cfda9a7d3bb2b991e790ea074566/podman/po.js","networkRequestId":"10303.67"}
CDP: {"source":"network","level":"error","text":"Failed to load resource: the server responded with a status of 404 (Not Found)","timestamp":1637665335006.557,"url":"http://127.0.0.2:9191/cockpit/$5d7b4a773812d7654b86f8c74ff42a11c495cfda9a7d3bb2b991e790ea074566/podman/po.js","networkRequestId":"10303.89"}
> warning: grep: /sys/class/dmi/id/power/autosuspend_delay_ms: Input/output error

time="2021-11-23T11:02:16Z" level=warning msg="The input device is not a TTY. The --tty and --interactive flags might not work properly"
Traceback (most recent call last):
  File "/usr/lib64/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/work/bots/make-checkout-workdir/test/check-application", line 959, in testCheckpointRestoreCGroupsV2
    return self._testCheckpointRestore()
  File "/work/bots/make-checkout-workdir/test/check-application", line 1054, in _testCheckpointRestore
    b.wait(lambda: self.getContainerAttr("swamped-crate", "State") in 'running')
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 401, in wait
    raise Error('timed out waiting for predicate to become true')
testlib.Error: timed out waiting for predicate to become true
Wrote screenshot to TestApplication-testCheckpointRestoreCGroupsV2-fedora-34-127.0.0.2-2301-FAIL.png
Wrote HTML dump to TestApplication-testCheckpointRestoreCGroupsV2-fedora-34-127.0.0.2-2301-FAIL.html
Wrote JS log to TestApplication-testCheckpointRestoreCGroupsV2-fedora-34-127.0.0.2-2301-FAIL.js.log
Journal extracted to TestApplication-testCheckpointRestoreCGroupsV2-fedora-34-127.0.0.2-2301-FAIL.log.gz
Warning: Stopping podman.service, but it can still be activated by:
  podman.socket
> warning: transport closed: disconnected
Failed to get session path: No session '66' known
Traceback (most recent call last):
  File "/work/bots/make-checkout-workdir/test/check-application", line 959, in testCheckpointRestoreCGroupsV2
    return self._testCheckpointRestore()
  File "/work/bots/make-checkout-workdir/test/check-application", line 1054, in _testCheckpointRestore
    b.wait(lambda: self.getContainerAttr("swamped-crate", "State") in 'running')
  File "/work/bots/make-checkout-workdir/test/common/testlib.py", line 401, in wait
    raise Error('timed out waiting for predicate to become true')
testlib.Error: timed out waiting for predicate to become true

# Result testCheckpointRestoreCGroupsV2 (__main__.TestApplication) failed
# 1 TEST FAILED [96s on 2-cockpit-10]
not ok 4 /work/bots/make-checkout-workdir/test/check-application TestApplication.testCheckpointRestoreCGroupsV2 [ND@1]
```
First occurrence: 2021-11-23T11:03:46.507560 | revision cb9438497c6042913af1da90966f01b5f120fc7b
Times recorded: 1
Latest occurrences:

- 2021-11-23T11:03:46.507560 | revision cb9438497c6042913af1da90966f01b5f120fc7b

**Comment by jelly** [comment]
This is now fixed in crun 1.4
